### ISSUE TITLE
Server crash System.NullReferenceException

### ISSUE THREAD

**Issue by Helidude** [N1Ran/BlockLimiter#12]
Message me on Discord if you need complete logs (Helidude)

> Torch.Managers.PluginManager: Loading plugin 'BlockLimiter' (v1.6.802)
.....
11:11:46.2522 [FATAL]  Initializer: System.NullReferenceException: Object reference not set to an instance of an object.
   at BlockLimiter.Utility.Block.RemoveBlock(MySlimBlock block) in D:\Sync\ProgramFiles\VisualStudio\source\repos\BlockLimiter\BlockLimiter\Utility\Block.cs:line 478
   at Sandbox.MySandboxGame.ProcessInvoke()
   at Sandbox.MySandboxGame.Update()
   at Sandbox.Engine.Platform.Game.UpdateInternal()
   at Sandbox.Engine.Platform.Game.RunSingleFrame()
   at Sandbox.Engine.Platform.FixedLoop.<>c__DisplayClass11_0.<Run>b__0()
   at Sandbox.Engine.Platform.GenericLoop.Run(VoidAction tickCallback)
   at Sandbox.Engine.Platform.Game.RunLoop()
   at Sandbox.MySandboxGame.Run(Boolean customRenderLoop, Action disposeSplashScreen)
   at Torch.VRageGame.DoStart() in C:\jenkins\workspace\Torch_Torch_master\Torch\VRageGame.cs:line 245
   at Torch.VRageGame.Run() in C:\jenkins\workspace\Torch_Torch_master\Torch\VRageGame.cs:line 125
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
System.NullReferenceException: Object reference not set to an instance of an object.
   at BlockLimiter.Utility.Block.RemoveBlock(MySlimBlock block) in D:\Sync\ProgramFiles\VisualStudio\source\repos\BlockLimiter\BlockLimiter\Utility\Block.cs:line 478
   at Sandbox.MySandboxGame.ProcessInvoke()
   at Sandbox.MySandboxGame.Update()
   at Sandbox.Engine.Platform.Game.UpdateInternal()
   at Sandbox.Engine.Platform.Game.RunSingleFrame()
   at Sandbox.Engine.Platform.FixedLoop.<>c__DisplayClass11_0.<Run>b__0()
   at Sandbox.Engine.Platform.GenericLoop.Run(VoidAction tickCallback)
   at Sandbox.Engine.Platform.Game.RunLoop()
   at Sandbox.MySandboxGame.Run(Boolean customRenderLoop, Action disposeSplashScreen)
   at Torch.VRageGame.DoStart() in C:\jenkins\workspace\Torch_Torch_master\Torch\VRageGame.cs:line 245
   at Torch.VRageGame.Run() in C:\jenkins\workspace\Torch_Torch_master\Torch\VRageGame.cs:line 125
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
11:11:46.2522 [INFO]   Initializer: Generating minidump at D:\Servers\AWorldOfClang\Instance\Minidump.dmp
11:11:46.2522 [ERROR]  Initializer: Keen broke the minidump, sorry.
11:11:51.5151 [INFO]   SteamCMD: Checking for DS updates.
11:11:57.3575 [INFO]   SteamCMD: Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...OK.

**Comment by N1Ran** [comment]
null reference should fixed on 1.6.804.
Let me know if you're still experiencing this

**Comment by Blaylock1988** [comment]
I believe I just experienced this crash:
```
18:14:23.6645 [FATAL]  Initializer: System.NullReferenceException: Object reference not set to an instance of an object.
   at BlockLimiter.Patch.ProjectionPatch.PrefixNewBlueprint(MyProjectorBase __instance, List`1& projectedGrids) in C:\Users\Olatide\Sync\ProgramFiles\VisualStudio\source\repos\BlockLimiter\BlockLimiter\Patch\ProjectionPatch.cs:line 73
   at Patched_Sandbox.Game.Entities.Blocks.MyProjectorBaseOnNewBlueprintSuccess_0(Object , List`1 )
   at VRage.Network.MyReplicationLayerBase.InvokeLocally[T1,T2,T3,T4,T5,T6,T7](CallSite`7 site, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7)
   at VRage.Network.MyReplicationLayer.DispatchEvent[T1,T2,T3,T4,T5,T6,T7,T8](CallSite callSite, EndpointId recipient, Nullable`1 position, T1& arg1, T2& arg2, T3& arg3, T4& arg4, T5& arg5, T6& arg6, T7& arg7, T8& arg8)
   at VRage.Network.MyReplicationLayerBase.RaiseEvent[T1,T2,T3](T1 arg1, T3 arg3, Func`2 action, T2 arg2, EndpointId endpointId, Nullable`1 position)
   at Sandbox.Engine.Multiplayer.MyMultiplayer.RaiseEvent[T1,T2](T1 arg1, Func`2 action, T2 arg2, EndpointId targetEndpoint)
   at Sandbox.Game.Entities.Blocks.MyProjectorBase.SendNewBlueprint(List`1 projectedGrids)
   at Sandbox.Game.Entities.Blocks.MyProjectorBase.<>c__DisplayClass88_0.<InitFromObjectBuilder>b__1()
   at ParallelTasks.Parallel.RunCallbacks()
   at Sandbox.Game.World.MySession.Update(MyTimeSpan updateTime)
   at Sandbox.MySandboxGame.Update()
   at Sandbox.Engine.Platform.Game.UpdateInternal()
   at Sandbox.Engine.Platform.Game.RunSingleFrame()
   at Sandbox.Engine.Platform.FixedLoop.<>c__DisplayClass11_0.<Run>b__0()
   at Sandbox.Engine.Platform.GenericLoop.Run(VoidAction tickCallback)
   at Sandbox.Engine.Platform.Game.RunLoop()
   at Sandbox.MySandboxGame.Run(Boolean customRenderLoop, Action disposeSplashScreen)
   at Torch.VRageGame.DoStart() in C:\jenkins\workspace\Torch_Torch_master\Torch\VRageGame.cs:line 246
   at Torch.VRageGame.Run() in C:\jenkins\workspace\Torch_Torch_master\Torch\VRageGame.cs:line 125
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
System.NullReferenceException: Object reference not set to an instance of an object.
   at BlockLimiter.Patch.ProjectionPatch.PrefixNewBlueprint(MyProjectorBase __instance, List`1& projectedGrids) in C:\Users\Olatide\Sync\ProgramFiles\VisualStudio\source\repos\BlockLimiter\BlockLimiter\Patch\ProjectionPatch.cs:line 73
   at Patched_Sandbox.Game.Entities.Blocks.MyProjectorBaseOnNewBlueprintSuccess_0(Object , List`1 )
   at VRage.Network.MyReplicationLayerBase.InvokeLocally[T1,T2,T3,T4,T5,T6,T7](CallSite`7 site, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7)
   at VRage.Network.MyReplicationLayer.DispatchEvent[T1,T2,T3,T4,T5,T6,T7,T8](CallSite callSite, EndpointId recipient, Nullable`1 position, T1& arg1, T2& arg2, T3& arg3, T4& arg4, T5& arg5, T6& arg6, T7& arg7, T8& arg8)
   at VRage.Network.MyReplicationLayerBase.RaiseEvent[T1,T2,T3](T1 arg1, T3 arg3, Func`2 action, T2 arg2, EndpointId endpointId, Nullable`1 position)
   at Sandbox.Engine.Multiplayer.MyMultiplayer.RaiseEvent[T1,T2](T1 arg1, Func`2 action, T2 arg2, EndpointId targetEndpoint)
   at Sandbox.Game.Entities.Blocks.MyProjectorBase.SendNewBlueprint(List`1 projectedGrids)
   at Sandbox.Game.Entities.Blocks.MyProjectorBase.<>c__DisplayClass88_0.<InitFromObjectBuilder>b__1()
   at ParallelTasks.Parallel.RunCallbacks()
   at Sandbox.Game.World.MySession.Update(MyTimeSpan updateTime)
   at Sandbox.MySandboxGame.Update()
   at Sandbox.Engine.Platform.Game.UpdateInternal()
   at Sandbox.Engine.Platform.Game.RunSingleFrame()
   at Sandbox.Engine.Platform.FixedLoop.<>c__DisplayClass11_0.<Run>b__0()
   at Sandbox.Engine.Platform.GenericLoop.Run(VoidAction tickCallback)
   at Sandbox.Engine.Platform.Game.RunLoop()
   at Sandbox.MySandboxGame.Run(Boolean customRenderLoop, Action disposeSplashScreen)
   at Torch.VRageGame.DoStart() in C:\jenkins\workspace\Torch_Torch_master\Torch\VRageGame.cs:line 246
   at Torch.VRageGame.Run() in C:\jenkins\workspace\Torch_Torch_master\Torch\VRageGame.cs:line 125
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

**Comment by N1Ran** [comment]
That's a shitty projection.  Will add some lines to keep those from crashing servers

**Comment by N1Ran** [comment]
projection null fixed in 1.6.817
